sktime/transformations/panel/dictionary_based/_sfa.py

```diff
@@ -233,25 +268,44 @@ def transform(self, X, y=None, supplied_dft=None):
 
         return bags
 
-    def _transform_case(self, X, i, supplied_dft):
+    def _transform_case(self, X, supplied_dft=None):
         if supplied_dft is None:
-            dfts = self._mft(X[i, :])
+            dfts = self._mft(X)
         else:
-            dfts = supplied_dft[i]
+            dfts = supplied_dft
 
-        bag = defaultdict(int)
-        # bag = Dict.empty(key_type=typeof((100,100.0)),
-        #                  value_type=types.float64) \
-        #     if self.levels > 1 else \
-        #     Dict.empty(key_type=types.int64, value_type=types.int64)
+        if self.typed_dict:
+            bag = (
+                Dict.empty(
+                    key_type=types.UniTuple(types.int64, 2), value_type=types.uint32
+                )
+                if self.levels > 1
+                else Dict.empty(key_type=types.int64, value_type=types.uint32)
+            )
+        else:
+            bag = {}
 
         last_word = -1
         repeat_words = 0
-        words = np.zeros(dfts.shape[0], dtype=np.int64)
+        words = (
+            np.zeros(dfts.shape[0], dtype=np.int64)
+            if self.word_bits <= 64
+            else [0 for _ in range(dfts.shape[0])]
+        )
 
         for window in range(dfts.shape[0]):
-            word_raw = SFA._create_word(
-                dfts[window], self.word_length, self.alphabet_size, self.breakpoints
+            word_raw = (
+                SFA._create_word(
+                    dfts[window],
+                    self.word_length,
+                    self.alphabet_size,
+                    self.breakpoints,
+                    self.letter_bits,
+                )
+                if self.word_bits <= 64
+                else self._create_word_large(
```

Comment:
Why does it have to be split between create_word and create_word_large? Is _create_word_large not possible with numba ?

Reply:
As far as I am aware max number of bits for numpy/numba ints is 64. The functions could be identical, but turning it into a numba function will convert the input/output to a 64 bit int (or crash if it fails to do so).

Makes it rather ugly, but we know how many bits we need beforehand and can use the fast functions for existing classifiers.

Reply:
Ah, ok, so the method switches between a return type with 32 and 64 bits, I see :)

Reply:
The 64 bits for the numba functions and unlimited for the pure python one, but essentially yeah.

Reply:
One suggestion for readability and reducing the overhead:

Maybe replace it by:
```
bigram = (
                        self._create_bigram_words(
                        word_raw, words[window - self.window_size])
                    )
```

and introduce a wrapper function:

 ```
   def _create_bigram_words(self, word_raw, word):
        return SFA._create_bigram_word(
                    word,
                    word_raw,
                    self.word_bits,
                ) if self.max_bits <= 64 else self._create_bigram_word_large(
                    word,
                    word_raw,
                )
```

This way, it can also be used further down the code for skip-grams:

                     ```
   skip_gram = (
                            self._create_bigram_words(
                                word_raw, words[window - s * self.window_size])
                        )
```

Reply:
A similar trick can be used in `_shorten_case` to increase readability.

Reply:
Good idea.